### PR TITLE
Implementation of triton algo in lambda

### PR DESCRIPTION
**Pull request by fgarberson**

### CODE REVIEW

File: cognoa_screeners/CDCwithSilentTritonQuestionnaire/api_handler.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
-from __future__ import print_function
+#from __future__ import print_function
```
Comment (mrhalim2001) [inline cognoa_screeners/CDCwithSilentTritonQuestionnaire/api_handler.py]: remove

File: cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py [hunk 2]
```diff
@@ -14,6 +14,13 @@ class Screener:
 	triton_4plus_addon_questionnaire = []
 	all_supported_questions = []
 
+	def _get_main_question_id(self, question_id):
```
Comment (mrhalim2001) [inline cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py]: _get_question_radix?

File: cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py [hunk 3]
```diff
@@ -14,6 +14,13 @@ class Screener:
...
+		''' Give a question that has a version number, get the
+		base of the question without the version information.
+		Convention is that version number is separated by double
+		underscore '''
+
+		return (question_id.split('__'))[0]
```
Comment (mrhalim2001) [inline cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py]: Make this delimiter a static variable

File: cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py [hunk 4]
```diff
@@ -79,19 +93,24 @@ def _read_screener_questionnaire_from_file(self, filename):
...
+
 		#supporting a new (Triton) questionnaire means that some base (CDC) questions will be updated and other (Triton-specific) questions will be considered add-on questions
 		def _apply_new_questionnaire(base_questionnaire, new_questionnaire):
+
 			base_questionnaire_ids = [x.identifier for x in base_questionnaire]
+			#### The list of the main parts of the question only for matching purposes
+			main_base_questionnaire_ids = [self._get_main_question_id(x.identifier) for x in base_questionnaire]
 			addon_questionnaire = []
 			for question in new_questionnaire:
+				main_question_id = self._get_main_question_id(question.identifier)
```
Comment (mrhalim2001) [inline cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py]: question_radix

File: cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py (hunk 3 above)
Comment (fgarberson) [inline cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py]: Done

File: cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py (hunk 2 above)
Comment (fgarberson) [inline cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py]: Done

File: cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py (hunk 4 above)
Comment (fgarberson) [inline cognoa_screeners/CDCwithSilentTritonQuestionnaire/cognoa_screener.py]: Done

**Pull request merged**